### ISSUE TITLE
Use Let's Encrypt on devilbox

### ISSUE THREAD

**Issue by robsoned** [cytopia/devilbox#611]
#### ISSUE TYPE

 - Feature request

#### SUMMARY

Use Let's Encrypt on production for a single website at least.

#### Goal

I'm using devilbox on production,  and I've been struggling to install Let's Encrypt work on it. There's a third party project on the documentation that is supposed to do it, but its documentation is not very helpful.

Thanks a lot for the project!

**Comment by cytopia** [comment]
@robsoned I haven't tried letsencrypt myself that's why I've linked the third-party. At what steps are you struggling?

**Comment by robsoned** [comment]
Hello @cytopia!  I was trying to run a Laravel application on production. I tried to follow some guides but I don't remember exactly where the installation fails.

but this one seems to be the most accurate guide.
https://tecadmin.net/install-lets-encrypt-create-ssl-ubuntu/

I will try to reproduce the error this week and come back with more precise information about it. Thank you!

**Comment by creactiv** [comment]
I came over this in another project, not using devilbox, and found out, that you can use for this the certbot package in Ubuntu / Debian (this are the OS i am some kind of familiar with), Using nginx, there is also a separate package for certbot with nginx. I made it happen somehow with addning 

apt-get install python-certbot-nginx
and then: 

certbot --nginx

then it will list all of your domains you are running and gives you the option to create certs for all of them or only for some of them.  Even better: it also updates the conf files within nginx

Perhaps this might help a bit...

**Comment by science695** [comment]
FYI: the main way this works is by the certbot taking over the webserver and serving a token file while Let's Encrypt attempts to hit your server from the internet.

TLDR: it doesn't work easily if your docker containers don't have public dns or public accessibility (which most local dev environments do not).

**Comment by science695** [comment]
@cytopia You can install certbot, and run a command and it will place challenge files in the docroot, do a handshake with letsencrypt and then retrieve a certificate. If you ran this on startup, it would check expiration dates and renew if needed.

Here is a configuration reference for one of the modes that could be helpful:
https://certbot.eff.org/docs/using.html#webroot

**Comment by cytopia** [comment]
@science695 I can ship certbot with the PHP-FPM images. Additionally a how-to is already available via the docs from another user. As the rest of the feature goes a bit out of scope for a development environment, I will only accept PR's for the rest of the features. Will leave this open until `certbot` is shipped :)

**Comment by science695** [comment]
@cytopia Sorry, didn't mean to bump a stale topic, just saw the help wanted tag from earlier.

**Comment by cytopia** [comment]
`certbot` will be shipped with this release: https://github.com/devilbox/docker-php-fpm/pull/159

**Comment by cytopia** [comment]
`certbot` is now available in `v1.7.1`

**Comment by shivam5th** [comment]
Follow [this guide](https://forum.dokuwiki.org/d/18706-embed-dokuwiki-page-in-another-non-dokuwiki-web-page/12) to install Let's Encrypt SSL on Ubuntu 20.04
